src/neo/SmartContract/InteropService.Contract.cs

```diff
@@ -40,44 +41,46 @@ private static bool Contract_Create(ApplicationEngine engine)
                 contract = new ContractState
                 {
                     Id = engine.Snapshot.ContractId.GetAndChange().NextId++,
-                    Script = script,
+                    Script = script.ToArray(),
                     Manifest = ContractManifest.Parse(manifest)
                 };
 
                 if (!contract.Manifest.IsValid(hash)) return false;
 
                 engine.Snapshot.Contracts.Add(hash, contract);
-                engine.CurrentContext.EvaluationStack.Push(StackItem.FromInterface(contract));
+                engine.Push(StackItem.FromInterface(contract));
                 return true;
             }
 
             private static bool Contract_Update(ApplicationEngine engine)
             {
-                byte[] script = engine.CurrentContext.EvaluationStack.Pop().GetSpan().ToArray();
-                if (script.Length > 1024 * 1024) return false;
-                var manifest = engine.CurrentContext.EvaluationStack.Pop().GetString();
-                if (manifest.Length > ContractManifest.MaxLength) return false;
+                if (!engine.TryPop(out StackItem item0)) return false;
+                if (!engine.TryPop(out StackItem item1)) return false;
 
                 var contract = engine.Snapshot.Contracts.TryGet(engine.CurrentScriptHash);
                 if (contract is null) return false;
 
-                if (script.Length > 0)
+                if (!item0.IsNull)
```

Comment:
I think that we should check the size of the item (null should be 0)

```suggestion
                if (!item0.Size > 0)
```

Reply:
The size is checked in next line.

Reply:
But it's checked the max size, if the user send [0] it will write an empty script

Reply:
So what?

Reply:
I think that we should not allow `new byte[0]` scripts

Reply:
Why not? This does not seem to have any disadvantages.

Reply:
What is the use case of create an empty script, i think that it could only produce mistakes

Reply:
Fixed.